extensions/wikia/ImageReview/ImageReviewSpecialController.class.php

```diff
@@ -43,71 +89,98 @@ public function index() {
-			$this->logImageListCompleteness( 'warning' );
+			/* SUS-541 / Mix <mix@wikia.com> / scope: the following if block */
+			if ( count( $this->imageList ) < ImageReviewHelper::LIMIT_IMAGES ) {
+				$this->logImageListCompleteness( 'warning' );
+			}
 		}
 
-		if ( count( $this->imageList ) == 0 ) {
-			$do = array( 
-				self::ACTION_QUESTIONABLE	=> ImageReviewStatuses::STATE_QUESTIONABLE,
-				self::ACTION_REJECTED		=> ImageReviewStatuses::STATE_REJECTED,
-				'default'					=> ImageReviewStatuses::STATE_UNREVIEWED
-			);
-
-			if ( isset( $do[ $this->action ] ) ) {
-				$this->imageList = $this->helper->getImageList( $this->ts, $do[ $this->action ], $this->order );
-				$this->imageCount = $this->helper->getImageCount();
-			} else {
-				$this->imageList = $this->helper->getImageList( $this->ts, $do[ 'default' ], $this->order );
-				$this->imageCount = $this->helper->getImageCount( ImageReviewSpecialController::ACTION_UNREVIEWED, count( $this->imageList ) );
+		if ( count( $imageList ) == 0 ) {
+			// If we're looking at unreviewed and there aren't any new images by latest
+			// timestamp, clear the image counts before refreshing the image list.
+			if ( $this->action == self::ACTION_UNREVIEWED ) {
+				$this->helper->clearCachedImageCount();
 			}
-		} else {
-			$this->imageCount = $this->helper->getImageCount();
+
+			$imageList = $this->helper->getImageList(
+				$this->ts,
+				$this::NEW_STATE_FOR_ACTION[$this->action],
+				$this->order
+			);
 		}
 
-		$this->setVariables();
+		return $imageList;
+	}
 
-		$this->imageCount = array_map(function($number) {
-			return $this->wg->Lang->formatNum($number);
-		}, $this->imageCount);
+	private function getImageCount() {
+		$imageCount = $this->helper->getImageCount();
 
-		/* SUS-541 / Mix <mix@wikia.com> / scope: the following if block */
-		$severity = count( $this->imageList ) < ImageReviewHelper::LIMIT_IMAGES ? 'error' : 'success';
-		$this->logImageListCompleteness( $severity );
+		// Format the number locally (add ',' or '.')
+		return array_map(
+			function( $number ) { return $this->wg->Lang->formatNum($number); },
+			$imageCount
+		);
 	}
 
-	public function stats() {
-		if ( !$this->wg->User->isAllowed( 'imagereviewstats' )) {
-			$this->specialPage->displayRestrictionError( 'imagereviewstats' );
-			return false;
-		}
+	private function requestedTimestampIsNewest() {
+		return $this->ts > $this->getNewestTimestamp();
+	}
+
+	private function getNewestTimestamp() {
+		$userKey = $this->helper->getUserTsKey();
+		return $this->wg->Memc->get( $userKey );
+	}
 
+	private function setNewestTimestampToCurrent() {
+		$userKey = $this->helper->getUserTsKey();
+		$this->wg->Memc->set( $userKey, $this->ts, 3600 /* 1h */ );
+	}
+
+	public function stats() {
 		$startDay = $this->request->getVal( 'startDay', date( 'd' ) );
 		$startMonth = $this->request->getVal( 'startMonth', date( 'n' ) );
 		$startYear = $this->request->getVal( 'startYear', date( 'Y' ) );
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "startYear" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)